File: Tutorials/#1 Get response in JSON format/return_json.py
import os
from openai import OpenAI
client = OpenAI()
OpenAI.api_key = os.getenv('OPENAI_API_KEY')

completion = client.chat.completions.create(
  model="gpt-4-1106-preview",
  messages=[
    {"role": "system", "content": "You are a helpful assistant. Your response should be in JSON format."},
    {"role": "user", "content": "Hello!"}
  ],
  response_format={"type": "json_object"}
)

print(completion.choices[0].message.content)

# Check if the OpenAI API response is a valid JSON
import json

def is_json(myjson):
  try:
    json.loads(myjson)
  except ValueError as e:
    return False
  return True

print(is_json(completion.choices[0].message.content))

File: openai-tutorials/1-Get_response_in_JSON_format/return_json.js
const OpenAI = require("openai");
const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

async function main() {
  const completion = await openai.chat.completions.create({
    model: "gpt-4-1106-preview",
    messages: [
      {
        role: "system",
        content:
          "You are a helpful assistant. Your response should be in JSON format.",
      },
      { role: "user", content: "Hello!" },
    ],
    response_format: { type: "json_object" },
  });

  console.log(completion.choices[0].message.content);

  // Check if the OpenAI API response is a valid JSON
  const isJSON = (obj) => {
    try {
      JSON.parse(obj);
      return true;
    } catch (e) {
      return false;
    }
  };

  console.log(isJSON(completion.choices[0].message.content));
}

main();
